src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.GuardedCallsTests.cs

```diff
@@ -210,10 +210,10 @@ class Target
 }" + MockAttributesCsSource + MockOperatingSystemApiSource;
 
             await VerifyAnalyzerAsyncCs(source,
-                VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.SupportedOsRule).WithLocation(15, 17).WithMessage("'Target.SupportedOnWindowsAndBrowser()' is supported on 'windows'"),
-                VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.SupportedOsVersionRule).WithLocation(16, 17).WithMessage("'Target.SupportedOnWindows10AndBrowser()' is supported on 'windows' 10.0 and later"),
-                VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.SupportedOsVersionRule).WithLocation(24, 17).WithMessage("'Target.SupportedOnWindows10AndBrowser()' is supported on 'browser'"),
-                VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.SupportedOsVersionRule).WithLocation(48, 17).WithMessage("'Target.SupportedOnWindows10AndBrowser()' is supported on 'browser'"));
+                VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.SupportedOsRule).WithLocation(15, 17).WithArguments("Target.SupportedOnWindowsAndBrowser()", "browser"),
+                VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.SupportedOsRule).WithLocation(16, 17).WithArguments("Target.SupportedOnWindows10AndBrowser()", "browser"),
```

Comment:
I would be surprised if this test passed. Shouldn't "10.0 and later" be passed as a third argument? and the rule should remain `SupportedOsVersionRule`?

Reply:
Well there is 2 diagnostics one for browser and the other for Windows 10.0 and later, as we can add only one markup for diagnostic the other is here, so there is nothing really change for the test

Reply:
@buyaa-n Does that mean when there are multiple diagnostics in the same span, the test need to only verify *one* of them?

Reply:
> @buyaa-n Does that mean when there are multiple diagnostics in the same span, the test need to only verify one of them?

Not test need to verify both of them, here one is verified by the markup syntax `[|ExpectedDiagnostic|]` in the source, i don't know if there is a way to mark there is 2 diagnostics, so the other one is added here as a parameter.

Reply:
@buyaa-n Got it. But if I understand that correctly, there is no way to test the **arguments** for both diagnostics. One of them is tested against its existence, and the other is tested against both existence and correctness of arguments.

There could be a room of improvement here in the tests infrastructure.

Reply:
> there is no way to test the arguments for both diagnostics. One of them is tested against its existence, and the other is tested against both existence and correctness of arguments.

You can do that for both, but for markup syntax you cannot verify the arguments inline, but you can number the markups and add corresponding arguments later for each markup. for this case for me it was enough to verify the message/location/arguments for only one of the diagnostics

Reply:
You can verify arguments using markup by using the markup syntax `{|#0:code|}` and then constructing the expected diagnostic with `.WithLocation(0).WithArguments(...)`. Multiple explicit diagnostics can reference the same location by identifier. You can also have two diagnostics at the same location by nesting: `[|[|code|]|]`, or `{|#0:{|#1:code|} more code|}`.

Reply:
As an aside, never use the two-argument form of `WithLocation`. Either use `WithLocation(int)` or use `WithSpan` to specify full spans.

Reply:
> As an aside, never use the two-argument form of `WithLocation`. Either use `WithLocation(int)` or use `WithSpan` to specify full spans.

I'll file an issue to improve this later.